modules/ossm-migrating-from-smcp-defined-gateways-to-gateway-injection.adoc

```diff
@@ -0,0 +1,142 @@
+====
++
+[source,yaml, subs="attributes,verbatim"]
+----
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: istio-ingressgateway-canary 
+  namespace: istio-system <1>
+spec:
+  selector:
+    matchLabels:
+      app: istio-ingressgateway
+      istio: ingressgateway 
+  template:
+    metadata:
+      annotations:
+        inject.istio.io/templates: gateway
+      labels: <2>
+        app: istio-ingressgateway
+        istio: ingressgateway
+        sidecar.istio.io/inject: "true"
+    spec:
+      containers:
+        - name: istio-proxy
+          image: auto
+      serviceAccountName: istio-ingressgateway
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: istio-ingressgateway
+  namespace: istio-system
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: secret-reader
+  namespace: istio-system
+rules:
+  - apiGroups: [""]
+    resources: ["secrets"]
+    verbs: ["get", "watch", "list"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: istio-ingressgateway-secret-reader
+  namespace: istio-system
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: secret-reader
+subjects:
+  - kind: ServiceAccount
+    name: istio-ingressgateway
+---
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy <3>
+metadata:
+  name: gatewayingress
+  namespace: istio-system
+spec:
+  podSelector:
+    matchLabels:
+      istio: ingressgateway
+  ingress:
+    - {}
+  policyTypes:
+    - Ingress
+---- 
+<1> The gateway injection deployment and all supporting resources should be deployed in the same namespace as the SMCP-defined gateway. 
+<2> Ensure that the labels specified in the pod template include all of the label selectors specified in the Service associated with the existing SMCP-defined gateway.
+<3> Grant access to the new gateway from outside the cluster. This access is required whenever the `spec.security.manageNetworkPolicy` of the `ServiceMeshControlPlane` resource is set to `true`, which is the default setting.
+
+. Verify that the new gateway deployment is successfully handling requests. 
++
+Ensure that the new gateway deployment is successfully handling requests. If access logging was configured in the `ServiceMeshControlPlane` resource view the access logs of the new gateway deployment to confirm the behavior.
+
+. Scale down the old deployment and scale up the new deployment. 
++
+You may gradually shift traffic from the old gateway deployment to the new gateway deployment by running the `oc scale` command at the command line:
++
+[source,terminal]
+----
+$ oc scale -n istio-system deployment/<gateway_deployment> --replicas <new_number_of_replicas>
+----
++
+As you scale, increase the number of replicas for the new gateway deployment and decrease the number of replicas for the old gateway deployment. Continue scaling until the new gateway deployment handles all traffic to the gateway Service. 
```

Comment:
I'm a little confused by the nature of this step. Is this meant to be optional? I don't think it is, but either way "You may" should probably be removed from the second sentence, since you're instructing users to run a command. If I'm interpreting this step right, where you repeat this command while increasing the replica values "until the new gateway deployment handles all traffic to the gateway Service", then I would suggest writing the step like this:

```suggestion
. Scale down the old deployment and scale up the new deployment. 
+
Gradually shift traffic from the old gateway deployment to the new gateway deployment by performing the following steps:

.. Increase the amount of replicas for the new gateway deployment by running the following command:
+
[source,terminal]
----
$ oc scale -n istio-system deployment/<new_gateway_deployment> --replicas <new_number_of_replicas>
----

.. Decrease the amount of replicas for the old gateway deployment by running the following command:
+
[source,terminal]
----
$ oc scale -n istio-system deployment/<old_gateway_deployment> --replicas <new_number_of_replicas>
----

 .. Run the previous two commands, each time increasing the number of replicas for the new gateway deployment and decreasing the number of replicas for the old gateway deployment, until the new gateway deployment handles all traffic to the gateway Service. 
```

This is just one way to rework this step, but the point is that I think breaking this down into specific, granular steps, where each step only has one explicit action for users to follow, would make this clearer for users to follow IMO. There are probably other ways to rework this, so let me know if you'd like to discuss this more!

Reply:
done